### PR TITLE
Configure auto-removal of jobs on completion/failure

### LINKED ISSUE #342: Clean-up old BullMQ jobs automatically
BullMQ won't automatically remove old jobs when they complete or fail.  We have to [configure it to do this automatically](https://docs.bullmq.io/guide/queues/auto-removal-of-jobs).

We should probably [remove all jobs after 7 days](https://docs.bullmq.io/guide/queues/auto-removal-of-jobs#keep-jobs-based-on-their-age) or something

### PR DESCRIPTION
**Pull request by SerpentBytes**
## Description
Resolves #342 by removing jobs on completion and failure after 7 days.

### CODE REVIEW

File: app/queues/certificate/certificate-flow.server.ts [hunk 1]
```diff
@@ -28,6 +28,11 @@ interface AddCertRequest {
   username: string;
 }
 
+const { JOB_REMOVAL_FREQUENCY_MS } = process.env;
+
+// constant  for removing job on completion/failure (in milliseconds)
+const JOB_REMOVAL_INTERVAL_MS = 7 * 24 * 60 * 60 * 1000; // 7 days
```
Comment (Genne23v) [inline app/queues/certificate/certificate-flow.server.ts:34]: 7 days seem very long. Does it have to be this long?

File: app/queues/certificate/certificate-flow.server.ts [hunk 2]
```diff
@@ -67,6 +72,8 @@ export const addCertRequest = async ({ rootDomain, username }: AddCertRequest) =
         type: 'exponential',
         delay: 60_000, // start with 1 minute, double each time
       },
+      removeOnComplete: { age: Number(JOB_REMOVAL_FREQUENCY_MS) || JOB_REMOVAL_INTERVAL_MS },
+      removeOnFail: { age: Number(JOB_REMOVAL_FREQUENCY_MS) || JOB_REMOVAL_INTERVAL_MS },
```
Comment (Genne23v) [inline app/queues/certificate/certificate-flow.server.ts:76]: `age` seems not ms as I read the documentation as below. 
https://docs.bullmq.io/guide/queues/auto-removal-of-jobs
```js
await myQueue.add(
  'test',
  { foo: 'bar' },
  {
    removeOnComplete: {
      age: 3600, // keep up to 1 hour
      count: 1000, // keep up to 1000 jobs
    },
    removeOnFail: {
      age: 24 * 3600, // keep up to 24 hours
    },
  },
);
```

And I think without argument name, `JOB_REMOVAL_INTERVAL_MS` will be the max number of jobs to keep.

**Review by Genne23v — CHANGES_REQUESTED** [review]

File: app/queues/certificate/certificate-flow.server.ts (hunk 1 above)
Comment (SerpentBytes) [reply app/queues/certificate/certificate-flow.server.ts:34]: I am not sure. I followed the suggested period in the issue description.

File: app/queues/certificate/certificate-flow.server.ts (hunk 2 above)
Comment (SerpentBytes) [reply app/queues/certificate/certificate-flow.server.ts:76]: > `age` seems not ms as I read the documentation as below. https://docs.bullmq.io/guide/queues/auto-removal-of-jobs
> 
> ```js
> await myQueue.add(
>   'test',
>   { foo: 'bar' },
>   {
>     removeOnComplete: {
>       age: 3600, // keep up to 1 hour
>       count: 1000, // keep up to 1000 jobs
>     },
>     removeOnFail: {
>       age: 24 * 3600, // keep up to 24 hours
>     },
>   },
> );
> ```

Yes, age needs to be in seconds; thanks for pointing that out. However, could you please clarify what you mean by:
>And I think without argument name, JOB_REMOVAL_INTERVAL_MS will be the max number of jobs to keep.

Comment (Genne23v) [reply app/queues/certificate/certificate-flow.server.ts:76]: This is from the doc 

### Keep a certain number of jobs

It is also possible to specify a maximum number of jobs to keep. A good practice is to keep a handful of completed jobs and a much larger value of failed jobs:

```
await myQueue.add(
  'test',
  { foo: 'bar' },
  { removeOnComplete: 1000, removeOnFail: 5000 },
);
```

Comment (Genne23v) [reply app/queues/certificate/certificate-flow.server.ts:76]: Sorry. Nevermind.I read the code wrong.

**Comment by SerpentBytes** [comment]
@Genne23v I have updated the value to be seven days in seconds instead of milliseconds; thanks for catching that!

**Review by Genne23v — APPROVED** [review]

File: .env.example [hunk 3]
```diff
@@ -32,10 +32,10 @@ APP_URL=http://host.docker.internal:8080
 SAML_IDP_METADATA_PATH=config/idp-metadata-dev.xml
 
 # Background jobs
-# 24 * 60 * 60 * 1000 = 604800000 (24 hours)
+# 24 * 60 * 60 * 1000 = 604800000 (24 hours in milliseconds)
 EXPIRATION_REPEAT_FREQUENCY_MS=86400000
-# 7 * 24 * 60 * 60 * 1000 = 604800000 (7 days)
-JOB_REMOVAL_FREQUENCY_MS=604800000
+# 7 * 24 * 60 * 60 * 1000 = 604800 (7 days in seconds)
+JOB_REMOVAL_FREQUENCY_S=604800
```
Comment (humphd) [inline .env.example:38]: Above you use `_MS`, let's not mix the unit in this file. Change this to `_MS` too.

Comment (humphd) [reply .env.example:38]: Also, please update `DEPLOY.md` with whatever you add to the app's environment variables

**Review by humphd — CHANGES_REQUESTED** [review]
Looks good.  One small thing + docs.

**Comment by SerpentBytes** [comment]
@humphd, I have updated the code based on your feedback. Reviews would be appreciated.

**Review by Genne23v — APPROVED** [review]

**Review by humphd — APPROVED** [review]

**Pull request merged**